### ISSUE TITLE
Rabbit MQ QueueDeclare for user with no "queue.declare" permissions

### ISSUE THREAD

**Issue by ascariandrea** [n8n-io/n8n#8419]
**Describe the bug**
The RabbitMQ Trigger node throws an error when using a user with no "queue.declare" permissions, cause the [implementation calls](https://github.com/n8n-io/n8n/blob/master/packages/nodes-base/nodes/RabbitMQ/GenericFunctions.ts#L78) "assertQueue" before connecting to the queue.

This is the error I get:
```
ERROR: Operation failed: QueueDeclare; 403 (ACCESS-REFUSED) with message "ACCESS_REFUSED - access to queue 'xxx-test' in vhost 'staging' refused for user 'n8n'"
```

**To Reproduce**
1 - Configure your RMQ user without "queue.declare" permission
2 - Create the queue in RMQ and give user READ access
3 - Configure you RMQ Trigger Node with the created queue
4 - Run the worflow
5 - Expect the error above

**Expected behavior**
I would expect an option to enable/disable the assertion of queues (and possibly exchanges) for such cases where n8n user has granted only "read" permissions.

**Environment (please complete the following information):**

- OS: Ubuntu Linux 22.04
- n8n Version 1.17.1
- Node.js Version 18.16.0
- Database system SQLite
- Operation mode main

**Comment by Joffcom** [comment]
Hey @ascariandrea,

This is an interesting one and I am not sure if this is a bug or a feature request, We use Assert to make sure the queue is there but I can also see why you wouldn't want to auto create queues.

Do you believe this is a bug or a feature request?

**Comment by ascariandrea** [comment]
Thank you @Joffcom, I actually think this could be a feature request (and applied to both _exchange_ and _queue_ definitions).

If you want me, I can open a PR in the next days 👍🏽

**Comment by Joffcom** [comment]
Hey @ascariandrea,

If you wanted to tackle it I am more than happy to review it, My initial thought was maybe a node option that uses assert by default and if disabled we can use checkQueue to make sure it exists before trying anything.

**Comment by ascariandrea** [comment]
Hey @Joffcom, 
I just opened the #8430, including the option to disable the _assertion_ on exchanges too! Let me know what you think :)

**Comment by Joffcom** [comment]
Nice work 👍🏻

**Comment by janober** [comment]
Fix got released with [`n8n@1.28.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.28.0)
